crates/rune/src/runtime/stack.rs

```diff
@@ -323,6 +323,14 @@ impl Stack {
         Ok(self.drain(count)?.collect::<Vec<_>>())
     }
 
+    /// Swap the value at position a with the value at position b.
+    pub(crate) fn swap(&mut self, a: usize, b: usize) -> Result<(), StackError> {
+        let a = self.stack.len().checked_sub(a).ok_or(StackError)?;
+        let b = self.stack.len().checked_sub(b).ok_or(StackError)?;
+        self.stack.swap(a, b);
```

Comment:
Yeah, that's necessary to ensure stack isolation between function calls, which could otherwise pose a security risk in case there are any miscompilations. Note that you get that for free if you checked_add from `stack_bottom` instead, all though you still need to check that neither exceeds the size of the stack to avoid the panic when out of bounds during `slice::swap`.

Reply:
Ok, thanks for your work!